### DISCUSSION TITLE
N64 retroblue not fighter friendly

### DISCUSSION THREAD

**Discussion by SinisterSticks** [darthcloud/BlueRetro#630] [Q&A] [answered]
I understand why you made the controllers map the way you did, but my joystick does not have an right analog controller therefore not all the buttons are accessible on my fight stick being plugged into an N64 retro blue adapter.

The instruction booklet mentioned something about Google Chrome and something that appears to look like button reassignments but it doesn't specifically say on the instruction booklet printed with the item I bought on eBay new in.box.

One, is there a button remap mode on these devices?  Two A, if yes how do I access it.  Two B, if no can you add that feature?

**Comment by darthcloud** [comment]
Ask the person you gave money to on eBay or read the wiki.

**Reply by SinisterSticks** [reply]
Look at my receipt.  It was actually Amazon where I purchased it from.  Any instruction booklet that came with the device said to talk about this on github.com so I assume this is the right place.

There is no right stick on my fight stick at least not without turning off the d-pad or the left stick and constantly turning it on and off to attack and move.

The sales pitch seems to imply some sort of config mode but I don't see or understand how to unlock it or use it.

It does have something to say about Google Chrome.

I thought the USB cable was just a power cable to keep it powered.  Is that supposed to be the interface with either an Android phone or an iPhone or a PC or a Mac?

**Comment by SinisterSticks** [comment]
Never mind I saw the Retroblue.io website as the web interface to use it with.

I understand do the nature of Open source it's not exactly the most intuitive consumer-level user friendly interface though you do fairly good attempts to trying to.

If Bluetooth Low Enrrgy mode is slow maybe a USB C male to USB C male cable would be good to plug it directly into a cell phone and you can access it more directly that way and hopefully quicker.

I have double female C connectors but not double male connectors for USB C.  Before I buy it and turn out to waste my money I would like to know if a USB C male to male would work on this to go faster vs bluetooth and web.  I read it takes 45 minutes to flash it via  Android and wireless methods.  What a wired connection to an Android phone make it significantly faster?

**Comment by SinisterSticks** [comment, marked as answer]
I think the N64 version of this should have an alternate fighter configuration for one main reason that is evident in the N64 that is not present in most of the other systems.

If you have a fight stick you have all digital buttons and you have all digital sticks.  When going back most systems the the buttons on the fight stick correspond to the buttons on a fight stick or a regular pad of an older system.  This is true of every system except one.

The Nintendo 64.  Your default mapping has no access to the heavy kick or the heavy punch from a control on the fight stick.  

On a regular N64 controller that defaults to C North and C east.

You guys do mirror what c west and see South is in a fight game which is Middle punch and Middle kick.

But you also have middle punch and middle kick mapped to the right analog stick assuming someone's going to play a twin stick game or normal pad game.

Most fights do not have access to the right analog stick.  Plus in theory if one does have access to the right analog stick on a hypothetical joystick which could be made with an Xbox adaptive controller the Xbox adapter controller restricts it to an analog axis from negative 1 to positive one.  

The reason why your systems in the N64 could get away with this is because there are enough buttons to accommodate to a fight stick and usually they are mapped correctly.

Systems younger than the N64 don't expect dual analog sticks.

And the  N64's is two main contemporaries, the PlayStation and Saturn both have the buttons of newer sticks, and the Saturn does not have a right analog stick and when the PlayStation does it doesn't pretend it's a button.

The corky thing about the N64 was it was trying to go for a right analog stick except it made for digital buttons for Super Mario 64.

Now I understand why the default mapping you would want the right stick to equal the four buttons in Super Mario 64.

But flicking your right stick to the right or down is not a fun way to play Mortal Kombat Trilogy or Killer Instinct gold.

That's because in some games it acts like a right analog stick and in other games it acts like extra action buttons.

By the way I have an Xbox adapter controller and if I wanted to rearrange buttons I could by using TRS 3.5 mm cables however C North and C East cannot be mapped to a button because the Adaptive controller expects a USB controller to represent the analog stick and it's expecting a stick not a button and it's expecting analog not digital.

So between most common fight sticks and the Xbox adaptive controller, It seems impossible to configure a setup to work with fight games.

Now I understand that you do have a config file but the problem is I can't figure out how to do it and it might be possible that I'm always going to be multiple minutes late because you said Android takes 45 minutes.  I'm trying to do stuff I don't understand on a Mac and it's taking 20 minutes and it went to the screensaver.  And I know on a map the screensaver interrupts what's going on internally sometimes.

That's why I request for the N64 to have fight stick mode so that people who have actual fight sticks and the Xbox adapter can actually use their fight sticks on Mortal Kombat Trilogy and Killer Instinct Gold.

By the way Brook has no adapter for the N64.  So unless Brooke makes an adapter we have to kind of rely on you.

Maybe I could raise money from fighters on Facebook and contribute one portion myself for you guys to add a fight stick mode for N64.

By the way if others have problems configuring the blue retro adapter you could just use an Xbox adaptive and just physically unplug and replug 3.5 mm TRS adapters.   You don't need the software we adjustment to do it.  It's a good thing I can or else I'm stuck trying to figure out how this works.
